internal/apex/wg-iface.go

```diff
@@ -0,0 +1,195 @@
+	if ifaceExists(logger, wgIface) {
+		_, err := RunCommand("ip", "link", "del", wgIface)
+		if err != nil {
+			logger.Debugf("failed to delete the ip link interface: %v\n", err)
+		}
+	}
+	// create the wireguard ip link interface
+	_, err := RunCommand("ip", "link", "add", wgIface, "type", "wireguard")
+	if err != nil {
+		logger.Errorf("failed to create the ip link interface: %v\n", err)
+	}
+	// start the wireguard listener on a well-known port if it is the hub-router as all
+	// nodes need to be able to reach this node for state distribution if hole punching.
+	if ax.hubRouter {
+		_, err = RunCommand("wg", "set", wgIface, "listen-port", strconv.Itoa(WgDefaultPort), "private-key", linuxPrivateKeyFile)
+		if err != nil {
+			logger.Errorf("failed to start the wireguard listener: %v\n", err)
+		}
+	} else {
+		// start the wireguard listener
+		_, err = RunCommand("wg", "set", wgIface, "listen-port", strconv.Itoa(ax.listenPort), "private-key", linuxPrivateKeyFile)
+		if err != nil {
+			logger.Errorf("failed to start the wireguard listener: %v\n", err)
+		}
+	}
+	// give the wg interface an address
+	_, err = RunCommand("ip", "address", "add", ax.wgLocalAddress, "dev", wgIface)
+	if err != nil {
+		logger.Debugf("failed to assign an address to the local linux interface, attempting to flush the iface: %v\n", err)
+		wgIP := getIPv4Iface(wgIface)
+		_, err = RunCommand("ip", "address", "del", wgIP.To4().String(), "dev", wgIface)
+		if err != nil {
+			logger.Errorf("failed to assign an address to the local linux interface: %v\n", err)
+		}
+		_, err = RunCommand("ip", "address", "add", ax.wgLocalAddress, "dev", wgIface)
+		if err != nil {
+			logger.Errorf("failed to assign an address to the local linux interface: %v\n", err)
+		}
+	}
+	// bring the wg0 interface up
+	_, err = RunCommand("ip", "link", "set", wgIface, "up")
+	if err != nil {
+		logger.Errorf("failed to bring up the wg interface: %v\n", err)
+	}
+}
+
+func setupWindowsIface(logger *zap.SugaredLogger, localAddress, privateKey string) error {
+	if err := buildWindowsWireguardIfaceConf(privateKey, localAddress); err != nil {
+		return fmt.Errorf("failed to create the windows wireguard wg0 interface file %v", err)
+	}
+
+	var wgOut string
+	var err error
+	if ifaceExists(logger, wgIface) {
+		wgOut, err = RunCommand("wireguard.exe", "/uninstalltunnelservice", wgIface)
+		if err != nil {
+			logger.Debugf("failed to down the wireguard interface (this is generally ok): %v\n", err)
+		}
+		time.Sleep(time.Second * 1)
+	}
+	logger.Debugf("stopped windows tunnel svc:%v\n", wgOut)
+	// sleep for one second to give the wg async exe time to tear down any existing wg0 configuration
+	wgOut, err = RunCommand("wireguard.exe", "/installtunnelservice", windowsWgConfigFile)
+	if err != nil {
+		return fmt.Errorf("failed to start the wireguard interface: %v\n", err)
+	}
+	logger.Debugf("started windows tunnel svc: %v\n", wgOut)
+	time.Sleep(time.Second * 1)
```

Comment:
Why is the final sleep necessary?

Reply:
I ran into a bit of delay in windows setting up the interface < 1s. Ran into an issue where it started trying to add peers before the iface was ready. Same scenario as above.